icefoganalytics/Travel-Authorization
# Travel Authorization Creation

Fixes https://github.com/icefoganalytics/Travel-Authorization/issues/2

# User Story

As a user, I want to create a new travel authorization request so that I can get approval for my travel.

I should be able to create these fields:
- travel steps
- travel method
- final destination
- purpose of travel

## Concerns

There seems to be some duplication in the UI components?
- `src/web/src/components/TravelRequest/TravelForm.vue`
- `src/web/src/modules/travelForm/views/TravelForm.vue`

# Testing Instruction

1. Boot the database service via `dev up`
2. Boot the api service via `cd src/api && npm run start`
3. Boot the web service via `cd src/web && npm run start`
4. Go to http://localhost:8080/my-travel-requests.
5. Click on the "+ Travel Authorization" button.
6. Complete the "Enter your personal details" section. No clue what the mailcode is supposed to be, but it doesn't seem to have any validation.
7. Complete the "Tell us about the travel" section.
8. Complete the "Enter details about the trip and purpose" section.
9. Click the "Submit to Supervisor" button.
10. There is no way, through the UI, for an end user to check what has been saved. However, if a developer has the console open, the response from the form submission endpoint is logged.
